gapis/gfxapi/gles/api/asynchronous_queries.api

```diff
@@ -85,13 +99,17 @@ cmd void glEndQuery(GLenum target) {
 @doc("https://www.khronos.org/opengles/sdk/docs/man31/html/glGenQueries.xhtml", Version.GLES31)
 @doc("https://www.khronos.org/opengles/sdk/docs/man32/html/glGenQueries.xhtml", Version.GLES32)
 cmd void glGenQueries(GLsizei count, QueryId* queries) {
+  GenQueries(count, queries)
+}
 
+sub void GenQueries(GLsizei count, QueryId* queries) {
+  if count < 0 { glErrorInvalidValue() }
   q := queries[0:count]
   ctx := GetContext()
   for i in (0 .. count) {
     id := as!QueryId(?)
-    // TODO: This should only reserve the name, but not create the object.
-    ctx.Objects.Queries[id] = new!Query(ID: id)
+    assert(id != 0)
+    ctx.Objects.GeneratedNames.Queries[id] = true
```

Comment:
All of the glGen* methods are pretty much identical, but I have not managed to figure out a nice way to avoid the copying pasting in the API language.
  cmd void glGenQueries(GLsizei count, QueryId* ids) {
    GenNames(ctx.Objects.GeneratedNames.Queries, count, ids)
  }
Maps are value objects, so this would not work, would it.  And there is no way to take a pointer in our language.  I miss C++